lib/common/logging.c

```diff
@@ -1148,14 +1153,33 @@ pcmk_log_xml_impl(uint8_t level, const char *text, const xmlNode *xml)
         }
 
         pcmk__output_set_log_level(logger_out, level);
+        pcmk__output_set_log_filter(logger_out, file, function, line, tags);
         pcmk__xml_show(logger_out, text, xml, 1,
                        pcmk__xml_fmt_pretty
                        |pcmk__xml_fmt_open
                        |pcmk__xml_fmt_children
                        |pcmk__xml_fmt_close);
+        pcmk__output_set_log_filter(logger_out, NULL, NULL, 0U, 0U);
```

Comment:
Alternatively, save and restore the original filters. However, it's hard to think of a valid use case for keeping a persistent filter.

Reply:
Yeah, there shouldn't be any reason to. I'll add a note to the doxygen that all callers should reset the filters when done.